### ISSUE TITLE
Crash playing videos from rumble.com in fullscreen

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#35817]
## Description 

Crash playing videos from rumble.com in fullscreen

Found while testing https://github.com/brave/brave-browser/issues/35721

## Steps to Reproduce

1. install `1.64.48`
2. launched Brave
3. loaded `rumble.com`
4. played a video
5. entered Picture-in-Picture (`PiP`)
6. entered fullscreen

## Actual result:

Crashes [`54f00c00-dba3-700b-0000-000000000000`]:

```
[ 0 ] VideoOverlayWindowViews::GetWorkAreaForWindow
[ 1 ] VideoOverlayWindowViews::OnDisplayMetricsChanged
[ 2 ] display::DisplayChangeNotifier::NotifyDisplaysChanged
[ 3 ] -[__NSObserver _doit:]
[ 4 ] -[NSNotificationCenter postNotificationName:object:userInfo:]
[ 5 ] notify_datagram_handler
[ 6 ] CGSDatagramReadStream::dispatchMainQueueDatagrams
[ 7 ] invocation function for block in CGSDatagramReadStream::mainQueueWakeup
```

## Expected result:

No crash, enters fullscreen video

## Reproduces how often: 

Often, but **_not_** 100%

## Brave version (brave://version info)

```
Brave | 1.64.48 Chromium: 121.0.6167.139 (Official Build) nightly (x86_64)
-- | --
Revision | 741f7ee622bdfde5e10e3beda02288f3c056287e
OS | macOS Version 11.7.10 (Build 20G1427)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

@sangwoo108 @temp-AD @rebron @brave/qa-team

**Comment by sangwoo108** [comment]
Looks like the same one with https://github.com/brave/brave-browser/issues/35528

When I saw it, it looks like the crash from the upstream code. Will check it out again. Thanks!

**Comment by bsclifton** [comment]
Should be fixed with https://github.com/brave/brave-core/pull/22468

@stephendonner can you please re-check? We can close as a duplicate if latest Nightly fixes issue 😄 Will need to be 1.65.60 or higher to verify

**Comment by stephendonner** [comment]
> Should be fixed with [brave/brave-core#22468](https://github.com/brave/brave-core/pull/22468)
> 
> @stephendonner can you please re-check? We can close as a duplicate if latest Nightly fixes issue 😄 Will need to be 1.65.60 or higher to verify

Yup, looks fixed/working with

```
Brave | 1.65.60 Chromium: 122.0.6261.111 (Official Build) nightly (x86_64)
-- | --
Revision | 340388685c2dcb4b42c7ac2fe0d765e536753dfe
OS | macOS Version 11.7.10 (Build 20G1427)
```
